fred2/initialstatus.cpp

```diff
@@ -438,17 +453,22 @@ void initial_status::OnOK()
 				if (hflag)
 					MODIFY(objp->hull_strength, (float) m_hull);
 
-				objp->flags.set(Object::Object_Flags::No_shields, m_has_shields == 0);
+				if (m_has_shields == 1)
+					objp->flags.remove(Object::Object_Flags::No_shields);
+				else if (!m_has_shields)
```

Comment:
More of a nitpick than anything, but I would use `else if (m_has_shields == 0)` to match the `if (m_has_shields == 1)`.